diff --git a/src/dxvk/dxvk_state_cache.cpp b/src/dxvk/dxvk_state_cache.cpp
index a564da8..761cbcd 100644
--- a/src/dxvk/dxvk_state_cache.cpp
+++ b/src/dxvk/dxvk_state_cache.cpp
@@ -1012,7 +1012,7 @@ namespace dxvk {
 
       for (uint32_t i = 0; i < numWorkers; i++) {
         m_workerThreads.emplace_back([this] () { workerFunc(); });
-        m_workerThreads[i].set_priority(ThreadPriority::Lowest);
+        m_workerThreads[i].set_priority(ThreadPriority::Normal);
       }
     }
   }
